Implementing RSpec in tic-tac-toe game project

o-rashad opened the pull request:
Merging feature/adding_tests branch to develop

Creating tests for public methods leads to create files :

- spec/player_spec.rb
- spec/game_spec.rb
- spec/public_spec.rb

o-rashad committed:
adding bin files

Rayan84 committed:
Add readme file

Rayan84 committed:
fixes on rubocop offenses

o-rashad committed:
fixing linters

o-rashad committed:
fixing workflow

o-rashad committed:
adding missing workflow files

o-rashad committed:
Fixing RSpec linters

o-rashad committed:
Merge branch 'feature/adding_tests' of https://github.com/Rayan84/tic-tac-toe into feature/adding_tests

o-rashad committed:
Fixing RSpec linters

o-rashad committed:
Fixing RSpec linters

PhillipUg reviewed (changes requested):
Hey team, 👋

You have done an amazing job implementing the project requirements of this project.

However, there are still some changes 👇 you need to implement to approve this request 🔍

### Changes Required ♻️ 

- [ ] You have some linter errors. Please always ensure you don't have any linter errors before submitting for review

- [ ] You have some failing specs. Please make sure that all your tests are passing before submitting for review.

- [ ] Also, when i run `RSpec`. it terminates prematurely and prompts me for input. This prevents it from running all your specs. Please fix this as well

#### Please, make all REQUIRED changes and submit your request again! 😀

If you have any questions, feel free to send me a message through Slack as **Phillip Musiime**
I would be happy to help! 👩‍💻 

Happy coding 🤓

o-rashad commented:
Thank you @PhillipUg for the review,
we were facing issues with the workflow files before the PR,
it wasn't indicating any linters or specs errors, 

later we addressed the issue and fixed the workflow directory.
now after the CR, we gonna work on it and push ASAP.

o-rashad committed:
Revert "fixing linters"

This reverts commit ea8b0951d8d94d4fa988c2076433a8f42ddd7b4b.

o-rashad committed:
Fixing linter and Update after 1st review

elmejdki reviewed (approved):
# status: approved :heavy_check_mark: 
Great job @team :clap: :clap: :clap:, thanks for making the required changes :+1: :100:.

now you can merge to develop your project is approved.

## Happy Coding :tophat:
